src/index.ts

```diff
@@ -733,50 +737,53 @@ export const plugin: PluginFunction<TypescriptMocksPluginConfig> = (schema, docu
-                                      prefix: config.prefix,
-                                      typesPrefix: config.typesPrefix,
-                                      enumsPrefix: config.enumsPrefix,
-                                      currentType: field.type,
-                                      customScalars: config.scalars,
-                                      transformUnderscore,
-                                      listElementCount,
-                                      dynamicValues,
-                                      generateLibrary,
-                                      generatorLocale,
-                                      fieldGeneration: config.fieldGeneration,
-                                      enumsAsTypes,
-                                      useTypeImports,
-                                      useImplementingTypes,
-                                      defaultNullableToNull,
-                                      nonNull: false,
-                                      typeNamesMapping: config.typeNamesMapping,
-                                  });
-
-                                  return `        ${field.name.value}: overrides && overrides.hasOwnProperty('${field.name.value}') ? overrides.${field.name.value}! : ${value},`;
-                              })
-                              .join('\n')
-                        : '';
+                    let mockFieldsString = '';
+
+                    const { directives } = node;
+                    const hasOneOfDirective = directives.some((directive) => directive.name.value === 'oneOf');
+
+                    if (node.fields && hasOneOfDirective) {
+                        const field = node.fields[0];
+                        const value = generateMockValue({
+                            typeName: fieldName,
+                            fieldName: field.name.value,
+                            currentType: field.type,
+                            ...sharedGenerateMockOpts,
+                        });
+
+                        mockFieldsString = `        ...(override ? override : {${field.name.value} : ${value}}),`;
+                    } else if (node.fields) {
+                        mockFieldsString = node.fields
+                            .map((field, index) => {
+                                const value = generateMockValue({
+                                    typeName: fieldName,
+                                    fieldName: field.name.value,
+                                    currentType: field.type,
+                                    ...sharedGenerateMockOpts,
+                                });
+
+                                const valueWithOverride = `overrides && overrides.hasOwnProperty('${
+                                    field.name.value
+                                }') ? overrides.${field.name.value}! : ${
+                                    !hasOneOfDirective || index === 0 ? value : 'undefined'
```

Comment:
`!hasOneOfDirective` is always true here as we are in the `else` statement.
So this can be updated to
```typescript
 const valueWithOverride = `overrides && overrides.hasOwnProperty('${field.name.value}') 
  ? overrides.${field.name.value}!
  : ${value}`;
```